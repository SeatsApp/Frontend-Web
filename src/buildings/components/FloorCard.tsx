import {
    Button,
    Card, Typography,
} from "@mui/material";
import { useNavigate } from "react-router-dom";
import { Floor } from "../../shared/types/Floor";

interface FloorCardProps {
    buildingId: number,
    floor: Floor
}

export default function FloorCard({ buildingId, floor }: FloorCardProps) {
    const navigate = useNavigate();

    return (
        <Card style={{ padding: 10, margin: 10 }}>
            <Typography>{floor.name}</Typography>
            <Button
                variant="contained"
                color={'secondary'}
                onClick={() => navigate("/buildings/" + buildingId)}
            >
                Change building
            </Button>
            <Button
                variant="contained"
                color={'secondary'}
                onClick={() => navigate("/buildings/" + buildingId + "/floors/" + floor.id + "/floorplan")}
            >
                Manage floorplan
            </Button>
        </Card>
    )
}