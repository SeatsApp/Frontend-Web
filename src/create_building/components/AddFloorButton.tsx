import { Box, Button } from "@mui/material";
import { Dispatch, SetStateAction } from "react";
import { Building } from "../../shared/types/Building";

interface AddFloorButtonProps {
    building: Building;
    setBuilding: Dispatch<SetStateAction<Building>>,
}

export default function AddFloorButton({ building, setBuilding }: AddFloorButtonProps) {

    const addFloor = () => {
        const floorCount = building.floors.length
        building.floors[floorCount] = {
            id: 0,
            name: "",
            points: [{
                firstPoint: 0,
                secondPoint: 0
            },
            {
                firstPoint: 1600,
                secondPoint: 0
            },
            {
                firstPoint: 1600,
                secondPoint: 1800
            },
            {
                firstPoint: 0,
                secondPoint: 1800
            },
            {
                firstPoint: 0,
                secondPoint: 0
            }],
            seats: []
        }
        setBuilding({ ...building, floors: building.floors })
    }

    return (
        <Box style={{ marginTop: 10, marginBottom: 10 }}>
            <Button variant="outlined" color={'secondary'}
                onClick={addFloor}>
                Add floor</Button>
        </Box>
    )
}