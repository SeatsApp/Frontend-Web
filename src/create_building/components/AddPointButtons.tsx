import { Box, Button } from "@mui/material";
import { Dispatch, SetStateAction } from "react";
import { Building } from "../../shared/types/Building";

interface AddPointButtonProps {
    floorIndex: number
    building: Building;
    setBuilding: Dispatch<SetStateAction<Building>>,
}

export default function AddPointButton({ floorIndex, building, setBuilding }: AddPointButtonProps) {

    const addPoint = (changeFloorIndex: number) => {
        const floor = building.floors[changeFloorIndex]
        const pointCount = floor.points.length
        floor.points[pointCount] = {
            firstPoint: 0,
            secondPoint: 0
        }
        building.floors[changeFloorIndex] = floor
        setBuilding({ ...building, floors: building.floors })
    }

    return (
        <Box style={{ marginTop: 10, marginBottom: 10 }}>
            <Button variant="contained" color={'secondary'}
                onClick={() => addPoint(floorIndex)}>
                Add point</Button>
        </Box>
    )
}