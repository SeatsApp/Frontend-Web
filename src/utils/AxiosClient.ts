import axios from "axios";
import {backendUrl} from "../config/EnvironmentConfig"

axios.defaults.withCredentials = true

const AxiosClient = axios.create({
    withCredentials: true,
    baseURL: backendUrl
})

AxiosClient.interceptors.request.use(async req => {
    const jwtToken = localStorage.getItem("JwtToken")
    if (req.headers !== undefined && jwtToken !== null) {
        req.headers.authorization = "Bearer " + jwtToken;
    }
    return req;
});

export default AxiosClient;