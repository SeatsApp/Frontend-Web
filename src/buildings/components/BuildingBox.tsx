import {
    Box, Typography,
} from "@mui/material";
import { Building } from "../../shared/types/Building";
import { Floor } from "../../shared/types/Floor";
import FloorCard from "./FloorCard";

interface BuildingBoxProps {
    building: Building
}

export default function BuildingBox({ building }: BuildingBoxProps) {
    return (
        <Box style={{ padding: 10, margin: 10 }}>
            <Typography variant='h5'>{building.name}</Typography>
            {building.floors.map((floor: Floor) =>
                <FloorCard key={floor.id} buildingId={building.id} floor={floor} />)}
        </Box>
    )
}