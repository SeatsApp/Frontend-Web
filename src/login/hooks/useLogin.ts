import { useState } from "react";
import AxiosClient from "../../utils/AxiosClient";

export default function useLogin() {
    const [loggedIn, setLoggedIn] = useState<boolean>(false);

    function checkLoggedIn() {
        return AxiosClient.get("/api/healthcheck")
            .then((response) => {
                if (response.status >= 300)
                    setLoggedIn(false)
                else
                    setLoggedIn(true)
            })
            .catch(() => {
                setLoggedIn(false)
            });
    }

    return {
        loggedIn,
        checkLoggedIn,
    };
}
