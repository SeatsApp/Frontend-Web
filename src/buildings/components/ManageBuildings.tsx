import useReadBuildings from "../hooks/useReadBuildings";
import {
    Box,
    Button,
    Card, Typography,
} from "@mui/material";
import BuildingBox from "./BuildingBox";
import { useNavigate } from "react-router-dom";
import { Building } from "../../shared/types/Building";


export const ManageBuildings = () => {
    const navigate = useNavigate();
    const { buildings } = useReadBuildings()

    return (
        <Box style={{ paddingTop: 20, margin: 2 }}>
            <Card style={{ padding: 20 }}>
                <Box style={{ display: "flex", flexWrap: 'wrap', justifyContent: 'space-between'}}>
                    <Typography color={'secondary'} variant='h3'>Buildings overview</Typography>
                    <Button
                        variant="contained"
                        color={'secondary'}
                        onClick={() => navigate("/buildings/0")}
                    >
                        Create building
                    </Button>
                </Box>
                {buildings.map((building: Building) =>
                    <BuildingBox key={building.id} building={building} />
                )}
            </Card>
        </Box>
    )
};