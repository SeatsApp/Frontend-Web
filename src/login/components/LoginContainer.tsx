import React, {useEffect} from "react"
import useLogin from "../hooks/useLogin";
import {
    Box, createTheme,
    CssBaseline,
    Grid,
    ThemeProvider,
} from "@mui/material";
import {OpenURLButton} from "./OpenUrlButton";
import {backendUrl} from "../../config/EnvironmentConfig";
import CronosLogin from "../../assets/cronosLogin.jpg";
import CronosBackground from "../../assets/cronosBackGround.png";

interface LoginContainerProps {
    children: JSX.Element;
}

export const LoginContainer = ({children}: LoginContainerProps) => {
    const {loggedIn, checkLoggedIn} = useLogin();

    useEffect(() => {
        const url = window.location.href;
        if (url !== null) {
            let jwtToken: string = url.split("=")[1]
            if (jwtToken !== undefined) {
                if (jwtToken.includes("#"))
                    jwtToken = jwtToken.substring(0, jwtToken.length - 1)
                localStorage.setItem("JwtToken", jwtToken)
            }
        }
        checkLoggedIn();
    }, []);

    const theme = createTheme();

    const loginUrl = backendUrl + "/api/admin/login";
    if (loggedIn) {
        return children
    } else {
        return (
            <ThemeProvider theme={theme}>
                <Grid container component="main" sx={{height: '100vh'}}>
                    <CssBaseline/>
                    <Grid
                        item
                        xs={false}
                        sm={4}
                        md={7}
                        sx={{
                            backgroundImage: `url(${CronosLogin})`,
                            backgroundRepeat: 'no-repeat',
                            backgroundColor: (t) =>
                                t.palette.mode === 'light' ? t.palette.grey[50] : t.palette.grey[900],
                            backgroundSize: 'cover',
                            backgroundPosition: 'center',
                        }}
                    />
                    <Grid item xs={12} sm={8} md={5} component={Box} sx={{
                        backgroundImage: `url(${CronosBackground})`,
                        backgroundRepeat: 'no-repeat',
                        backgroundSize: 'cover',
                        backgroundPosition: 'center',
                        display: 'flex',
                        flexWrap: 'nowrap',
                        justifyContent: 'center',
                        alignItems: 'center',
                    }}>
                        <Box sx={{height: '50%', display: 'flex', alignItems: 'center'}}>
                            <Box>
                                <OpenURLButton url={loginUrl}/>
                            </Box>
                        </Box>
                    </Grid>
                </Grid>
            </ThemeProvider>
        )
    }
}