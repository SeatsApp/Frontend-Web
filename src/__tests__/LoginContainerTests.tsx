import React from "react";
import renderer from "react-test-renderer";
import 'jest-localstorage-mock';
import {LoginContainer} from "../login/components/LoginContainer";
import {Typography} from "@mui/material";
import AxiosClient from "../utils/AxiosClient";
import useLogin from "../login/hooks/useLogin";

beforeEach(() => {
    jest.spyOn(console, 'warn').mockImplementation();
    jest.spyOn(console, 'error').mockImplementation();
});

jest.mock("../utils/AxiosClient");
const mockedAxios = AxiosClient as jest.Mocked<typeof AxiosClient>

test("renders not logged in correctly", () => {
    const tree = renderer.create(<LoginContainer>
        <Typography>Test</Typography>
    </LoginContainer>).toJSON();
    expect(tree).toMatchSnapshot();
});

test("get jwt token from url correctly", () => {
    const useStateSpy = jest.spyOn(React, 'useState');
    useStateSpy.mockImplementation(() => [false, jest.fn()])

    // @ts-ignore
    delete window.location
    // @ts-ignore
    Object.defineProperty(window, 'location', {
        value: {
            href: "https://testurl?=#1234"
        }
    });

    mockedAxios.get.mockResolvedValue({
        status: 200
    });

    const tree = renderer.create(<LoginContainer>
        <Typography>Test</Typography>
    </LoginContainer>).toJSON();

    expect(tree).toMatchSnapshot();
    expect(localStorage.getItem("JwtToken")).toEqual("#123");
});

test("renders logged in correctly", () => {
    const setState = jest.fn();
    const useStateSpy = jest.spyOn(React, 'useState');
    useStateSpy.mockImplementation(() => [true, setState])

    mockedAxios.get.mockResolvedValue({
        status: 200
    });

    const tree = renderer.create(<LoginContainer>
        <Typography>Test</Typography>
    </LoginContainer>).toJSON();
    expect(tree).toMatchSnapshot();
});

test("should call api with correct parameters", async () => {
    const setState = jest.fn();
    const useStateSpy = jest.spyOn(React, 'useState');
    useStateSpy.mockImplementation(() => [false, setState])

    const mockAxios = jest.spyOn(AxiosClient, 'get')
    mockAxios.mockImplementation(() => Promise.resolve({
        status: 200
    }));

    const { checkLoggedIn } = useLogin();

    await checkLoggedIn();

    expect(setState).toBeCalledWith(true);
});


test("should call api with wrong parameters", async () => {
    const setState = jest.fn();
    const useStateSpy = jest.spyOn(React, 'useState');
    useStateSpy.mockImplementation(() => [false, setState])

    const mockAxios = jest.spyOn(AxiosClient, 'get')
    mockAxios.mockImplementation(() => Promise.resolve({
        status: 302
    }));

    const { checkLoggedIn } = useLogin();

    await checkLoggedIn();

    expect(setState).toBeCalledWith(false);
});

