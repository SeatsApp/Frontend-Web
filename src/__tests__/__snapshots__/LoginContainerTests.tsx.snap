// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders logged in correctly 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
>
  Test
</p>
`;

exports[`renders not logged in correctly 1`] = `
<main
  className="MuiGrid-root MuiGrid-container css-1jrn2fe-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-7 css-bxq4xw-MuiGrid-root"
  />
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation6 MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-5 css-gadse4-MuiPaper-root-MuiGrid-root"
  >
    <div
      className="MuiBox-root css-0"
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "height": "50%",
          "justifyContent": "center",
          "position": "relative",
        }
      }
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        style={
          Object {
            "fontSize": 20,
            "margin": 20,
            "textAlign": "center",
          }
        }
      >
        You have to login with your cronos account.
      </p>
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Login
        </p>
      </button>
    </div>
  </div>
</main>
`;
